tests/utils/test_enumerableset.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+from starkware.starknet.testing.starknet import Starknet
+from utils import contract_path, TRUE, FALSE
+
+
+@pytest.fixture
+async def enumerableset_mock():
+    starknet = await Starknet.empty()
+
+    enumerableset = await starknet.deploy(
+        contract_path("tests/mocks/enumerableset_mock.cairo")
+    )
+
+    return enumerableset
+
+
+@pytest.mark.asyncio
+async def test_add(enumerableset_mock):
+    enumerableset = enumerableset_mock
+    
+    executed_info = await enumerableset.length().call()
+    assert executed_info.result == (0,)
+    
+    success = await enumerableset.add(1).invoke()
+    assert success.result.success == TRUE
+
+    executed_info = await enumerableset.contains(1).call()
+    assert executed_info.result == (TRUE,)
+
+    executed_info = await enumerableset.length().call()
+    assert executed_info.result == (1,)
+
+
+@pytest.mark.asyncio
+async def test_remove(enumerableset_mock):
+    enumerableset = enumerableset_mock
+    success = await enumerableset.add(1).invoke()
+    assert success.result.success == TRUE
+
+    success = await enumerableset.remove(1).invoke()
+    assert success.result.success == TRUE
+
+    executed_info = await enumerableset.length().call()
+    assert executed_info.result == (0,)
+
+
+@pytest.mark.asyncio
+async def test_remove_not_present(enumerableset_mock):
+    enumerableset = enumerableset_mock
+
+    success = await enumerableset.remove(1).invoke()
+    assert success.result.success == FALSE
+
+    executed_info = await enumerableset.contains(1).call()
+    assert executed_info.result == (FALSE,)
+
+
+@pytest.mark.asyncio
+async def test_dupe_add(enumerableset_mock):
+    enumerableset = enumerableset_mock
+
+    success = await enumerableset.add(1).invoke()
+    assert success.result.success == TRUE
+
+    success = await enumerableset.add(1).invoke()
+    assert success.result.success == FALSE
+
+    executed_info = await enumerableset.contains(1).call()
+    assert executed_info.result == (TRUE,)
+
+    executed_info = await enumerableset.length().call()
+    assert executed_info.result == (1,)
+
+
+@pytest.mark.asyncio
+async def test_multi_add(enumerableset_mock):
+    enumerableset = enumerableset_mock
+    
+    for i in [1,3,2]:
+        success = await enumerableset.add(i).invoke()
+        assert success.result.success == TRUE
+
+    for i in [3,2,1]:
+        executed_info = await enumerableset.contains(i).call()
+        assert executed_info.result == (TRUE,)
+
+    executed_info = await enumerableset.length().call()
+    assert executed_info.result == (3,)
+
+
+@pytest.mark.asyncio
+async def test_add_three_remove_one(enumerableset_mock):
+    enumerableset = enumerableset_mock
+
+    for i in [1,3,2]:
+        success = await enumerableset.add(i).invoke()
+        assert success.result.success == TRUE
```

Comment:
Since this method is repeated in a few other tests, we could create a fixture to avoid the repetition, and use pytest's parametrize with the integers/booleans to consolidate this test (and the few others like it) to a few lines. I think this is what @martriay was referring to...maybe? 🤷‍♂️

It will tighten up these last few tests. I think these examples might be helpful:

Fixturization: [https://github.com/OpenZeppelin/cairo-contracts/blob/main/tests/token/erc721_enumerable/test_ERC721_Enumerable_Mintable_Burnable.py#L74-L84](https://github.com/OpenZeppelin/cairo-contracts/blob/main/tests/token/erc721_enumerable/test_ERC721_Enumerable_Mintable_Burnable.py#L74-L84)

Parametrization: [https://github.com/OpenZeppelin/cairo-contracts/blob/main/tests/token/erc721/test_ERC721_Mintable_Burnable.py#L887-L899](https://github.com/OpenZeppelin/cairo-contracts/blob/main/tests/token/erc721/test_ERC721_Mintable_Burnable.py#L887-L899)

Reply:
yes